### ISSUE TITLE
Error: Invalid hook call

### ISSUE THREAD

**Issue by fnarbona-adesso** [Nazeeh21/Chakra-UI-Carousel#7]
**Code file contents:**

```
import { Box, Text } from '@chakra-ui/react';
import { Provider, Carousel } from 'chakra-ui-carousel';

const Slider = () => {
	return (
		<Box>
			<Provider>
				<Carousel gap={50}>
					<Box>
						<Text>Slide 1</Text>
					</Box>
					<Box>
						<Text>Slide 2</Text>
					</Box>
					<Box>
						<Text>Slide 3</Text>
					</Box>
				</Carousel>
			</Provider>
		</Box>
	);
};

export default Slider;
```

**Terminal details:**

```
wait  - compiling...
event - compiled client and server successfully in 1142 ms (745 modules)
warn  - Fast Refresh had to perform a full reload. Read more: https://nextjs.org/docs/basic-features/fast-refresh#how-it-works
wait  - compiling / (client and server)...
warn  - Fast Refresh had to perform a full reload. Read more: https://nextjs.org/docs/basic-features/fast-refresh#how-it-works
event - compiled client and server successfully in 353 ms (982 modules)
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1476:13)
    at Object.useState (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1507:20)
    at Provider (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/dist/chakra-ui-carousel.cjs.dev.js:19:51)
    at renderWithHooks (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)
    at renderIndeterminateComponent (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)
    at renderElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)
    at renderNodeDestructiveImpl (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)
    at renderNodeDestructive (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)
    at renderNode (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6259:12)
    at renderHostElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5642:3)
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1476:13)
    at Object.useState (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1507:20)
    at Provider (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/dist/chakra-ui-carousel.cjs.dev.js:19:51)
    at renderWithHooks (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)
    at renderIndeterminateComponent (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)
    at renderElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)
    at renderNodeDestructiveImpl (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)
    at renderNodeDestructive (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)
    at renderNode (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6259:12)
    at renderHostElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5642:3)
error - Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1476:13)
    at Object.useState (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1507:20)
    at Provider (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/dist/chakra-ui-carousel.cjs.dev.js:19:51)
    at renderWithHooks (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)
    at renderIndeterminateComponent (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)
    at renderElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)
    at renderNodeDestructiveImpl (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)
    at renderNodeDestructive (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)
    at renderNode (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6259:12)
    at renderHostElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5642:3) {
  page: '/'
}
error - Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1476:13)
    at Object.useState (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/node_modules/react/cjs/react.development.js:1507:20)
    at Provider (/Users/fernandonarbona/Repos/adesso-website/node_modules/chakra-ui-carousel/dist/chakra-ui-carousel.cjs.dev.js:19:51)
    at renderWithHooks (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5658:16)
    at renderIndeterminateComponent (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5731:15)
    at renderElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5946:7)
    at renderNodeDestructiveImpl (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6104:11)
    at renderNodeDestructive (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6076:14)
    at renderNode (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:6259:12)
    at renderHostElement (/Users/fernandonarbona/Repos/adesso-website/node_modules/react-dom/cjs/react-dom-server.browser.development.js:5642:3) {
  page: '/'
}
```

**Browser message:**

<img width="981" alt="image" src="https://user-images.githubusercontent.com/116831172/200436350-bab75b66-853d-438a-832b-24c55e78b6f6.png">

**Comment by ensarbayhan** [comment]
Same issue here. Is there any way to fix this issue?

**Comment by Frisbeetarian** [comment]
Facing same issue.

**Comment by dalemugford** [comment]
Same issue..

**Comment by with-heart** [comment]
It would be very helpful if one of you could provide a minimal codesandbox reproduction of the issue

**Comment by fnarbona-adesso** [comment]
> It would be very helpful if one of you could provide a minimal codesandbox reproduction of the issue

@with-heart Just a basic implementation per the instructions fails. You don't need much to reproduce. In any case, here is a link to code sandbox [https://codesandbox.io/s/chakra-ui-carousel-issue-8drs4i?file=/src/App.js](url)

**Comment by davidcrab** [comment]
same issue

**Comment by wes** [comment]
same issue

**Comment by calnighters** [comment]
Same issue here

**Comment by rodrigojacomeliRID** [comment]
Pessoal, passei por este problema aqui também, porém consegui resolver colocando o Provider como o primeiro componente que encapsula todo o restante. segue exemplo ...

`export function BannersCarousel() {
  return (
    <Provider>
      <Carousel gap={50}>
        <Image
          src='#'
          alt='Banner'
        />

        <Image
          src='#'
          alt='Banner'
        />
      </Carousel>
      <LeftButton />
      <RightButton />
    </Provider>
  );
}`

Espero ter ajudado!

**Comment by rodrigojacomeliRID** [comment]
Pessoal, também passei por este problema aqui em uma aplicação em NextJS, mas consegui fazer funcionar da seguinte forma.

1. Criar um componente a parte para o Carousel.
2. O elemento Provider da lib, tem que ser o primeiro elemento no qual encapsula todo o restante.
3. Exportando o componente criado com já de costume com demais criações de componentes.

Segue um exemplo do código que aqui em meu projeto funcionou;

![BannerCarousel_Fix](https://user-images.githubusercontent.com/79519519/219016268-d23ad8c7-8dec-4b1d-85a7-422a5022d464.png)

Após isso é só importar o componente criado anteriormente com a lib e importar dentro de outro componente como desejado;

![BannerCarousel_Fix_02](https://user-images.githubusercontent.com/79519519/219016795-6ab71e6d-8ba2-4cba-b29e-083836478134.png)


Espero ter ajudado...

**Comment by vidvidvid** [comment]
same

**Comment by naz3eh** [comment]
sorry to hear that! I'll have a look at it soon and will push a fix for it soon.

**Comment by naz3eh** [comment]
hey guys! I've looked into the error and it seems that this error is on the latest version of nextjs. I will find a fix for it and will soon push it. Sorry for the inconvenience.

**Comment by naz3eh** [comment]
Extremely sorry for the inconvenience caused to you guys. I've pushed the fix for this issue, and it works perfectly now. Refer to the below video for reference.

https://user-images.githubusercontent.com/56908732/222560227-0ec4820c-2619-49bc-9783-e387cd00ea43.mov

**Comment by naz3eh** [comment]
Closed by #9
